DEVELOPMENT.md

```diff
@@ -77,15 +77,24 @@ make test
 
 ### Building
 
-You can build the stack with the following command:
+There is a single binary for the server, `ttn-lw-stack`, as well as a binary for the command-line interface `ttn-lw-cli`. The single binary contains all components start one or multiple components. This allows you to run the stack with one command in simple deployment scenarios, as well as distributing micro-services for more advanced scenarios.
+
+We provide binary releases for all supported platforms, including packages for various package managers at https://github.com/TheThingsNetwork/lorawan-stack/releases . We suggest you use the compiled packages we provide in production scenarios.
+
+For development/testing purposes we suggest either running required binaries via `go run`(e.g. `go run ./cmd/ttn-lw-cli` from repository root for CLI), or using `go build` directly.
```

Comment:
Now that we no longer recommend `make clean build-all` we should also give instructions on how to build the frontend during development, right? CC: @bafonins @kschiffer 

Can be a separate PR as far as I'm concerned.